src/Factory/OrderFactory.php

```diff
@@ -94,4 +137,29 @@ private function getCustomerForOrder(string $email): CustomerInterface
 
         return $customer;
     }
+
+    private function copyOrderItemsToReorder(OrderInterface $order, OrderInterface $reorder): void
+    {
+        $orderItems = $order->getItems();
+
+        /** @var OrderItemInterface $orderItem */
+        foreach ($orderItems as $orderItem) {
+            /** @var ProductVariantInterface $productVariant */
+            $productVariant = $orderItem->getVariant();
+            if ($productVariant->isTracked() && !$productVariant->isInStock()) {
+                continue;
+            }
+
+            /** @var OrderItemInterface $newItem */
+            $newItem = $this->orderItemFactory->createNew();
+
+            $newItem->setVariant($orderItem->getVariant());
+            $newItem->setUnitPrice($orderItem->getUnitPrice());
+            $newItem->setProductName($orderItem->getProductName());
+            $newItem->setVariantName($orderItem->getVariantName());
+
+            $this->orderItemQuantityModifier->modify($newItem, $orderItem->getQuantity());
+            $this->orderModifier->addToOrder($reorder, $newItem);
```

Comment:
Maybe in the nearest future we can extract some services for order creation, as this class has lots of dependencies :(